File: scripts/open_app.py
import argparse
import subprocess
import shlex
import json
import os
from desktop_actions import DesktopAction
from pathlib import Path

# check for cl args
parser = argparse.ArgumentParser("open_app")
parser.add_argument("app_name", help="the app being opened", type=str)
parser.add_argument("installed_apps", help="where installed app info is serialized", type=str)
args = parser.parse_args()

# read installed apps
with open(args.installed_apps, "r") as json_file:
    apps = json.load(json_file)

# if not apps[args.app_name]["exec"]:
#     raise ValueError("no exec value found!")

try:
    env_vars = os.environ.copy()
    env_vars["GTK_PATH"] = ""
    launch_app_cmd = f"gio launch {apps[args.app_name]['.desktop']}"
    installed_apps_cmd = subprocess.run(shlex.split(launch_app_cmd), capture_output=True, check=True, encoding='utf-8', env=env_vars)
except subprocess.CalledProcessError as cpe:
    raise cpe

File: src/scripts/open-app.ts
import { exec } from "child_process"
import { promisify } from "util";
import { AppInfo } from "@/scripts/installed-apps";

const execAsync = promisify(exec)

export async function openApp(appsInfo: AppInfo, appName: string) {
    try {
        const launchAppCmd = `gio launch ${appsInfo[appName].desktopPath}`
        const launchApp = await execAsync(launchAppCmd)
        return launchApp.stdout
    } catch (launchError) {
        throw launchError
    }
} 